### ISSUE TITLE
Workflow automation is failing randomly on the "Evaluate rules" step

### ISSUE THREAD

**Issue by floernoult** [linear-b/gitstream#312]
**Describe the bug**
Hi,
We've configured Gitstream on our repo with only one automation taken from the examples : `estimated_time_to_review` 
The Github action itself is an exact copy of the website examples, with only a very light change to filter our bot.

What's strange is that it seems the action is run several times on every PR, even if on the PR you can only see it once in the list.
On the action itself, you can see the runs that are launched:
<img width="532" alt="Capture d’écran 2023-09-07 à 17 00 18" src="https://github.com/linear-b/gitstream/assets/1006512/7052a433-c3e4-4817-b5db-4a1d3d3c8d37">

But on the PR, you see one occurence in the checklist:
<img width="827" alt="Capture d’écran 2023-09-07 à 17 07 44" src="https://github.com/linear-b/gitstream/assets/1006512/0b299f62-4417-4eea-892d-90731d8c09ec">

And sometimes they all pass, sometimes they almost all fail.
Let me know if you need anything else.
Thanks

**To Reproduce**

1. The `.cm` automation file
```
# -*- mode: yaml -*-

manifest:
  version: 1.0

automations:
  estimated_time_to_review:
    if:
      - true
    run:
      - action : add-label@v1
        args:
          label: "{{ calc.etr }} min review"
          color: {{ 'E94637' if (calc.etr >= 20) else ('FBBD10' if (calc.etr >= 5) else '36A853') }}

calc:
  etr: {{ branch | estimatedReviewTime }}
```

2. The PR URL (as it contains the repo and PR identifiers)
--> A recent example is https://github.com/agorapulse/platform/pull/29457

3. Describe your PR relevant content 
--> This PR contains only backend code (dependencies update), but we're having the issue with any kind of PRs (containing frontend code, backend code, tests, refactorings, etc..)

**Expected behavior**

We expect the action to be run only once per PR and to pass, especially since the tag with the estimated time to review is correctly applied.

**Screenshots**
<img width="1336" alt="Capture d’écran 2023-09-07 à 16 55 27" src="https://github.com/linear-b/gitstream/assets/1006512/2a5ae197-50ac-4d37-be01-5b06006c31da">
<img width="748" alt="Capture d’écran 2023-09-07 à 16 55 52" src="https://github.com/linear-b/gitstream/assets/1006512/58e51aaa-91aa-4fed-ae09-afe2f966785f">

**Comment by vim-zz** [comment]
@floernoult is it a new issue you have? if yes, since when did you notice it?

**Comment by floernoult** [comment]
Hi @vim-zz 
No it's not new. My team reported me the issue during the summer as it's starting to be a pain for most of our developers, but it has been months that it's happening. I could find traces of failures for this action every month since January. The action is not required to merge the PRs so it never blocked us, but I think it got worst in the past two month and having so many PRs' automations being red in Github is an issue.

I was wondering if the fact that gitstream action is running 4 or 5 times concurrently when a PR is created could cause this issue? I understand that this action is triggered when a PR is created or updated, and maybe upon creation there are several updates on the PR, made by other actions.

**Comment by vim-zz** [comment]
There is a new feature that allows you to control the triggers, see https://docs.gitstream.cm/execution-model/#explicit-triggers

you can try adding the following to reduce the number of runs:

```
on:
  - pr_created
  - commit
```

**Comment by floernoult** [comment]
Thanks for the tip. I just updated our `cm` file, I'll see how it behaves today and let you know.

**Comment by floernoult** [comment]
Hi @vim-zz 
Things are looking way better since I updated the script. I had only 2 failures, with no more concurrent runs. And those failures have other reasons
```
Error: {
  "message": "Git command failed. reason: Command failed: cd repo && git diff $'master'...$'upstream/chore/sc-108289-fix-payload' -- .cm/*.cm\nfatal: bad revision 'master...upstream/chore/sc-108289-fix-payload'\n"
}
```

**Comment by floernoult** [comment]
Hi @vim-zz 
Limiting the execution of the actions helped a lot. I still see some failures but not for the same reason. So I'm closing the issue for now.
Thanks for your help
